modules/globebrowsing/src/geojson/globegeometryfeature.cpp

```diff
@@ -0,0 +1,918 @@
+
+        shader->setUniform("heightOffset", _offsets.z);
+        shader->setUniform("useHeightMapData", useHeightMap());
+
+        if (shader == _linesAndPolygonsProgram) {
+            shader->setUniform("normalTransform", normalTransform);
+            shader->setUniform("nLightSources", lightSourceData.nLightSources);
+            shader->setUniform("lightIntensities", lightSourceData.intensitiesBuffer);
+            shader->setUniform(
+                "lightDirectionsViewSpace",
+                lightSourceData.directionsViewSpaceBuffer
+            );
+        }
+
+        glBindVertexArray(r.vaoId);
+
+        switch (r.type) {
+            case RenderType::Lines:
+                shader->setUniform(
+                    "opacity",
+                    r.isExtrusionFeature ? fillOpacity : opacity
+                );
+                renderLines(r);
+                break;
+            case RenderType::Points:
+                shader->setUniform("opacity", opacity);
+                renderPoints(r, renderData, pointRenderMode);
+                break;
+            case RenderType::Polygon: {
+                shader->setUniform("opacity", fillOpacity);
+                renderPolygons(r, shouldRenderTwice, pass);
+                break;
+            }
+            default:
+                throw ghoul::MissingCaseException();
+                break;
+        }
+
+        shader->deactivate();
+    }
+
+    glBindVertexArray(0);
+
+    // Reset after every geometry
+    global::renderEngine->openglStateCache().resetPolygonAndClippingState();
+}
+
+void GlobeGeometryFeature::renderPoints(const RenderFeature& feature,
+                                        const RenderData& renderData,
+                                        const PointRenderMode& renderMode) const
+{
+    ghoul_assert(feature.type == RenderType::Points, "Trying to render faulty geometry");
+    _pointsProgram->setUniform("color", _properties.color());
+    _pointsProgram->setUniform(
+        "pointSize",
+        0.001f * _properties.pointSize() * static_cast<float>(_globe.boundingSphere())
+    );
+
+    _pointsProgram->setUniform("renderMode", static_cast<int>(renderMode));
+
+    using TextureAnchor = GeoJsonProperties::PointTextureAnchor;
+    _pointsProgram->setUniform(
+        "useBottomAnchorPoint",
+        _properties.pointTextureAnchor() != TextureAnchor::Center
+    );
+
+    // Points are rendered as billboards
+    glm::dvec3 cameraViewDirectionWorld = -renderData.camera.viewDirectionWorldSpace();
+    glm::dvec3 cameraUpDirectionWorld = renderData.camera.lookUpVectorWorldSpace();
+    glm::dvec3 orthoRight = glm::normalize(
+        glm::cross(cameraUpDirectionWorld, cameraViewDirectionWorld)
+    );
+    if (orthoRight == glm::dvec3(0.0)) {
+        glm::dvec3 otherVector = glm::vec3(
+            cameraUpDirectionWorld.y,
+            cameraUpDirectionWorld.x,
+            cameraUpDirectionWorld.z
```

Comment:
A short comment why the y,x,z order?

Reply:
Ah..! Copied this from renderablebillboardscloud. Gotta figure out why the order is like that first 😅